nomad/server.go

```diff
@@ -1531,7 +1531,6 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string) (
 			return nil, err
 		}
 	}
-	conf.RejoinAfterLeave = true
```

Comment:
Oof, looks like this was left in from the original implementation: https://github.com/hashicorp/nomad/blame/main/nomad/server.go#L1534